client/views/view_units.cpp

```diff
@@ -17,26 +18,78 @@
 #include "tileset/sprite.h"
 #include "top_bar.h"
 
+// common
+#include "canvas.h"
+#include "goto.h"
+#include "movement.h"
+#include "text.h"
+
 /**
  * Constructor for units view
  */
 units_view::units_view() : QWidget()
 {
   ui.setupUi(this);
 
+  const QPixmap *spr;
+  QImage img;
+  QRect crop;
+  QImage cropped_img;
+  QPixmap pix;
   QStringList slist;
-  slist << _("Unit Type") << _("Upgradable ★") << _("In Progress ⚒")
-        << _("Active ⚔") << _("Shield Upkeep") << _("Food Upkeep")
+
+  // Configure the units table
+  slist << _("Unit Type") << _("★ Upgradable") << _("⚒ In Progress")
+        << _("⚔ Active") << _("Shield Upkeep") << _("Food Upkeep")
         << _("Gold Upkeep") << QLatin1String("");
+  ui.units_label->setText(QString(_("Units:")));
   ui.units_widget->setColumnCount(slist.count());
   ui.units_widget->setHorizontalHeaderLabels(slist);
   ui.units_widget->setSortingEnabled(false);
   ui.upg_but->setText(_("Upgrade"));
   ui.upg_but->setDisabled(true);
   ui.find_but->setText(_("Find Nearest"));
-  ui.find_but->setDisabled(false);
+  ui.find_but->setDisabled(true);
   ui.disband_but->setText(_("Disband All"));
-  ui.disband_but->setDisabled(false);
+  ui.disband_but->setDisabled(true);
+
+  // Configure the unitwaittime table
+  slist.clear();
+  slist << _("Type") << _("Location") << _("Mp") << _("Time left")
+        << QLatin1String("");
+  ui.uwt_widget->setColumnCount(slist.count());
+  ui.uwt_widget->setHorizontalHeaderLabels(slist);
+  ui.uwt_widget->setSortingEnabled(false);
+  ui.uwt_label->setText("Units Waiting:");
+
+  // Add shield icon for shield upkeep column
+  spr = tiles_lookup_sprite_tag_alt(tileset, LOG_VERBOSE, "upkeep.shield",
+                                    "citybar.shields", "", "", false);
+  img = spr->toImage();
+  crop = zealous_crop_rect(img);
+  cropped_img = img.copy(crop);
+  pix = QPixmap::fromImage(cropped_img);
+  ui.units_widget->horizontalHeaderItem(4)->setIcon(pix);
+
+  // Add food icon for food upkeep column
+  spr = tiles_lookup_sprite_tag_alt(tileset, LOG_VERBOSE, "citybar.food",
+                                    "citybar.food", "", "", false);
+  img = spr->toImage();
+  crop = zealous_crop_rect(img);
+  cropped_img = img.copy(crop);
+  pix = QPixmap::fromImage(cropped_img);
+  ui.units_widget->horizontalHeaderItem(5)->setIcon(pix);
+
+  // Add gold icon for gold upkeep column
+  // FIXME: We don't have a sprite for a gold coin on its own. This looks
```

Comment:
`upkeep.gold`?

Reply:
I don't think there is a sprite for that, hence the ask of hugomflavio for one.

Reply:
`upkeep.gold`, for amplio2 in upkeep.png:
![image](https://user-images.githubusercontent.com/22327575/218207423-3636a8d9-dc82-4c99-81fd-94f028501388.png)

Reply:
But that is only for amplio2, correct? Was looking for something tileset agnostic. The three in use are in `misc`.

Reply:
`upkeep.shield` is not, though the alt `citybar.shields` is (but since `upkeep.shields` is mandatory, the alt will never be used). In any case they can be overridden by the tileset by not importing `misc`.